app/src/main/java/org/mozilla/fenix/RunBlockingCounter.kt

```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// This class implements the alternative ways to invoke runBlocking with some
+// monitoring by wrapping the raw methods. This lint check tells us not to use the raw
+// methods so we suppress the check.
+@file:Suppress("MozillaRunblockingCheck")
+
+package org.mozilla.fenix
+
+import androidx.annotation.VisibleForTesting
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.runBlocking
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+
+/**
+ * Counts the number of runBlocking calls made
+ */
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+class RunBlockingCounter {
+    companion object {
+        var runBlockingCount = 0L
```

Comment:
This needs to be synchronized (if it's synchronized, does it automatically get volatile too? I'm not sure)

Reply:
If I add `@Synchronize set` for this variable, isn't there an issue where Thread 1 --> Increases counter, then runs its method, returns and Thread 2 --> waits for the method to return before increasing?   Although I don't think that would happen, since the lock would be released as soon as we're done incrementing. I guess the only issue would be if a billion threads queue'd up at the same time to increment the counter, which I think is unlikely.

Maybe I'm overthinking it though, I added the `@Synchronized` with my latest fix. I saw nothing locally but it was just a thought.

Reply:
Yes, that's possible. I'm concerned that the value will not be accessed correctly in concurrent situations, however.

Actually, I don't know how the java memory model works. Is assignment atomic? In that case, I don't think we need synchronized, just volatile. We could also change it to an `AtomicInteger` (or whatever) to to avoid having to figure it out.

Reply:
From what I remember, if the interger is `volatile` it is kept in the main memory and access to that integer is the same throughout different threads. I think, however, that `AtomicInteger` can solve this issue

Reply:
@mcomella  So I added `AtomicInteger` but I also think we need to `synchronized(...)` on the check for overflow since two threads my overlap. what do you think?